src/Microsoft.AspNet.Mvc.Core/UrlHelper.cs

```diff
@@ -120,8 +120,13 @@ protected virtual string GeneratePathFromRoute(string routeName, IDictionary<str
         }
 
         /// <inheritdoc />
-        public virtual string Content([NotNull] string contentPath)
+        public virtual string Content(string contentPath)
         {
+            if (contentPath == null)
+            {
+                return null;
+            }
+
             return GenerateClientUrl(_httpContext.Request.PathBase, contentPath);
```

Comment:
Hmm, part of me feels like it'd be too protective over users; but then again I can't imagine a user purposefully calling into this method with an untrimmed value.

I'm not sure change this method directly would be the right approach; however, it brings up some gray area in the `UrlResolutionTagHelper` where it is smarter about parsing URLs with whitespace where our `ScriptTagHelper`, `LinkTagHelper` and `ImageTagHelper` take a users value verbatim and hand it over to the `IUrlHelper`.